ts/src/header-validator/source.test.ts

```diff
@@ -1927,6 +1930,289 @@ const testCases: TestCase[] = [
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "max_event_states": 4294967296
+    }`,
+    expectedErrors: [
+      {
+        path: ['max_event_states'],
+        msg: 'must be in the range [1, 4294967295]',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scopes-size-exceeds-attribution-scope-limit',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "attribution_scopes": ["1", "2"]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'the size of attribution scopes must be smaller than attribution scope limit (1)',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-not-string',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scopes": [1, 2]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes', 0],
+        msg: 'must be a string',
+      },
+      {
+        path: ['attribution_scopes', 1],
+        msg: 'must be a string',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scopes-empty-list',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "attribution_scopes": []
+    }`,
+  },
+  {
+    name: 'attribution-scopes-not-list',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "attribution_scopes": 1
+    }`,
+    expectedErrors: [
+      {
+        msg: 'must be a list',
+        path: ['attribution_scopes'],
+      },
+    ],
+  },
+  {
+    name: 'attribution-scopes-too-many',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "attribution_scopes": ["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21"]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [0, 20]',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scopes-too-long',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "attribution_scopes": ["aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes', 0],
+        msg: 'exceeds max length per attribution scope (51 > 50)',
+      },
+    ],
+  },
+  {
+    name: 'channel-capacity-attribution-scope-event',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 10,
+      "max_event_states": 15
+    }`,
+    sourceType: SourceType.event,
+    noteInfoGain: true,
+    vsv: {
+      maxEventLevelChannelCapacityPerSource: {
+        [SourceType.event]: 6.5,
+        [SourceType.navigation]: 11.5,
+      },
+      maxSettableEventLevelEpsilon: 14,
+    },
+    expectedErrors: [
+      {
+        path: [],
+        msg: 'information gain: 7.11 exceeds max event-level channel capacity per event source (6.50)',
```

Comment:
Would it be possible to add some context to this error message to indicate that the issue is related to the usage of scopes?

Something like:

```
information gain for max_event_states: ...
```